Add missing SQA doco for python packages

GiudGiud opened the pull request:
1300 warnings -> 1100 left
refs #8661 #7451 #15968

GiudGiud committed:
Rename a chigger testing folder that works on global not field (spatially dependent) data

GiudGiud committed:
Remove duplicated files in chigger simple tests

GiudGiud committed:
Add missing test for timings of the plotting script
Rework testing script to work in a clean repository

GiudGiud committed:
Add testing SQA items for chigger
refs #8661

GiudGiud committed:
Add testing SQA items to python Parameter objects

GiudGiud committed:
Add testing SQA items for peacock
refs #7451

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2114063/) on 088b45c wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/27070/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2114078/) on 088b45c wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/stochastic_tools)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/27070/coverage/xfem)

</details>



This comment will be updated on new commits.

GiudGiud committed:
Make Python SQA requirements strict
refs #15968

cticenhour commented:
@GiudGiud added to my list for review tomorrow.

cticenhour commented on python/chigger/tests/viewports/tests:
```diff
@@ -1,8 +1,11 @@
     command = viewports.py
     imagediff = 'viewports.png'
     display_required = true
+    requirement = 'The system shall be able to specify the coordinates of a view port in the results.'
```
```suggestion
    requirement = 'The system shall be able allow the user to specify the coordinates of a view port in the results.'
```

Should it say this instead? Not familiar with this feature.

cticenhour commented on python/chigger/tests/writers/tests:
```diff
@@ -1,8 +1,11 @@
     input = writers.py
     buffer = True
     display_required = true
+    requirement = 'The system shall be able to write plot with various formats, and error if an unsupported format is requested.'
```
```suggestion
    requirement = 'The system shall be able to write a plot with various formats, and error if an unsupported format is requested.'
```

cticenhour commented on python/parameters/test/tests:
```diff
@@ -1,10 +1,14 @@
   [Parameter]
     type = PythonUnitTest
     input = test_Parameter.py
+    requirement = "The system shall be able to define parameters in Python, similar to individual MooseObject parameters in C++, with numerous similar capabilities, including a name, a type, allowed and default values, descriptions, required and private status."
```
```suggestion
    requirement = "The system shall be able to define parameters and associated metadata in Python that allows for type-checking, restrictions, defaults, verification, and documentation."
```

cticenhour commented on python/parameters/test/tests:
```diff
@@ -1,10 +1,14 @@
   [InputParameters]
     type = PythonUnitTest
     input = test_InputParameters.py
+    requirement = "The system shall be able to define groups of parameters to Python objects, similar to the parameters of a MooseObject in C++, with numerous similar capabilities, including parameter errors, the ability to retrieve parameters, to set defaults, and others."
```
```suggestion
    requirement = "The system shall have a warehouse and capability to manage groups of parameters in Python objects, including parameter errors and the ability to retrieve and set parameters, to set defaults, and to validate parameters."
```

cticenhour commented on python/peacock/tests/base/tests:
```diff
@@ -1,12 +1,16 @@
     type = RunApp
     command = test_PeacockCollapsibleWidget.py
     display_required = true
+    requirement = 'The system shall have a graphical user interface which supports collapside widgets.'
```
```suggestion
    requirement = 'The system shall have a graphical user interface which supports collapsible widgets.'
```

cticenhour commented on python/peacock/tests/execute_tab/ConsoleOutputViewerPlugin/tests:
```diff
@@ -1,7 +1,10 @@
     type = PythonUnitTest
     input = test_ConsoleOutputViewerPlugin.py
     display_required = true
+    requirement = "The system shall include a graphical interface that display scrollable and auto-scrollable console output."
```
```suggestion
    requirement = "The system shall include a graphical interface that displays scrollable and auto-scrollable console output."
```

cticenhour commented on python/peacock/tests/execute_tab/ExecuteOptionsPlugin/tests:
```diff
@@ -1,7 +1,10 @@
     type = PythonUnitTest
     input = test_ExecuteOptionsPlugin.py
     display_required = true
+    requirement = "The system shall include a graphical interface that lets user specifies execution options such as the number of threads, parallel processes, working directory and input file name."
```
```suggestion
    requirement = "The system shall include a graphical interface that lets users specify execution options such as the number of threads, parallel processes, working directory and input file name."
```

cticenhour commented on python/peacock/tests/execute_tab/ExecuteTabPlugin/tests:
```diff
@@ -1,7 +1,10 @@
     type = PythonUnitTest
     input = test_ExecuteTabPlugin.py
     display_required = true
+    requirement = "The system shall include a graphical interface that has a plugin execution tab, that lets user specify execution parameters, through the command line, programmatically and on the interface, and set these run option preferences for individual plugins."
```
```suggestion
    requirement = "The system shall include a graphical interface that has a plugin execution tab that lets user specify execution parameters, through the command line, programmatically and on the interface, and set these run option preferences for individual plugins."
```

cticenhour commented on python/peacock/tests/exodus_tab/tests:
```diff
@@ -1,93 +1,110 @@
     input = test_VTKWindowPlugin.py
     separate = True
     display_required = true
+    requirement = "The system shall include a graphical interface that can display both mesh and solution Exodus files and let the user, including series of files created with mesh adaptivity."
```
Let the user what?

GiudGiud replied:
Added a list.
These tests are very not-granular. I m sure it saved some time building the test suite but it makes for off requirements

cticenhour commented on python/peacock/tests/input_tab/BlockEditor/tests:
```diff
@@ -1,7 +1,10 @@
     type = PythonUnitTest
     input = test_BlockEditor.py
     display_required = true
+    requirement = "The system shall include a graphical interface that lets the user modify input files for example by adding or cloning blocks, specifying their parameters, handling the types of the parameters, and handling regex 'star' syntax."
```
```suggestion
    requirement = "The system shall include a graphical interface that lets the user modify input files, for example, by adding or cloning blocks, specifying their parameters, handling the types of the parameters, and handling regex 'star' syntax."
```

cticenhour commented on python/peacock/tests/input_tab/BlockInfo/tests:
```diff
@@ -1,7 +1,10 @@
     type = PythonUnitTest
     input = test_BlockInfo.py
     display_required = true
+    requirement = "The system shall include a graphical interface that can gather and process parameters for input files blocks, notably their type, name, input path and value."
```
```suggestion
    requirement = "The system shall include a graphical interface that can gather and process parameters for input file blocks - notably their type, name, input path and value."
```

cticenhour commented on python/peacock/tests/input_tab/BlockTree/tests:
```diff
@@ -1,7 +1,10 @@
     type = PythonUnitTest
     input = test_BlockTree.py
     display_required = true
+    requirement = "The system shall include a graphical interface that can store an input file as a tree of blocks and lets user modify individual blocks."
```
```suggestion
    requirement = "The system shall include a graphical interface that can store an input file as a tree of blocks and lets users modify individual blocks."
```

cticenhour commented on python/peacock/tests/input_tab/CommentEditor/tests:
```diff
@@ -1,7 +1,10 @@
     type = PythonUnitTest
     input = test_CommentEditor.py
     display_required = true
+    requirement = "The system shall include a graphical interface that lets user specify comments in their input files."
```
```suggestion
    requirement = "The system shall include a graphical interface that lets users specify comments in their input files."
```

cticenhour commented on python/peacock/tests/input_tab/InputFile/tests:
```diff
@@ -1,7 +1,10 @@
     type = PythonUnitTest
     input = test_InputFile.py
     display_required = true
+    requirement = "The system shall include a graphical interface that can read user inputs and error if they are missing, present invalid duplicate syntax, or an unsupported file extension."
```
```suggestion
    requirement = "The system shall include a graphical interface that can read user inputs and error if they are missing, present invalid duplicate syntax, or are an unsupported file extension."
```

cticenhour commented on python/peacock/tests/input_tab/InputFileEditor/tests:
```diff
@@ -1,7 +1,10 @@
     type = PythonUnitTest
     input = test_InputFileEditor.py
     display_required = true
+    requirement = "The system shall include a graphical interface that lets user create new inputs files, including new blocks and suggesting missing required blocks."
```
```suggestion
    requirement = "The system shall include a graphical interface that lets users create new input files, including adding new blocks and suggesting missing required blocks."
```

cticenhour commented on python/peacock/tests/input_tab/InputFileEditorPlugin/tests:
```diff
@@ -1,7 +1,10 @@
     type = PythonUnitTest
     input = test_InputFileEditorPlugin.py
     display_required = true
+    requirement = "The system shall include a graphical interface that lets user modify input files inside widgets."
```
```suggestion
    requirement = "The system shall include a graphical interface that lets users modify input files inside widgets."
```

cticenhour commented on python/peacock/tests/input_tab/InputFileEditorWithMesh/tests:
```diff
@@ -1,8 +1,11 @@
     input = test_InputFileEditorWithMesh.py
     separate = True
     display_required = true
+    requirement = "The system shall include a graphical interface that lets user modify input files with an existing mesh, pulling in block information from the mesh and highlighting blocks and boundaries in the mesh when specifying them as parameters."
```
```suggestion
    requirement = "The system shall include a graphical interface that lets users modify input files with an existing mesh, pulling in block information from the mesh and highlighting blocks and boundaries in the mesh when specifying them as parameters."
```

cticenhour commented on python/peacock/tests/input_tab/OutputNames/tests:
```diff
@@ -1,7 +1,10 @@
     type = PythonUnitTest
     input = test_OutputNames.py
     display_required = true
+    requirement = "The system shall include a graphical interface that lets user specify new file names for the application output, and follows expected name conventions otherwise."
```
```suggestion
    requirement = "The system shall include a graphical interface that lets users specify new file names for the application output and follows default name conventions when a new name is not specified."
```

cticenhour commented on python/peacock/tests/input_tab/ParameterInfo/tests:
```diff
@@ -1,7 +1,10 @@
     type = PythonUnitTest
     input = test_ParameterInfo.py
     display_required = true
+    requirement = "The system shall include a graphical interface that can read and store parameter information, such as for example the parameter name, description, values, C++ and Python value types."
```
```suggestion
    requirement = "The system shall include a graphical interface that can read and store parameter information, such as, for example, the parameter name, description, values, C++ and Python value types."
```

cticenhour commented on python/peacock/tests/peacock_app/check_exodus_state/tests:
```diff
@@ -1,8 +1,11 @@
     input = test_exodus_state.py
     separate = True
     display_required = true
+    requirement = "The system shall include a graphical interface which maintains its exodus file viewing settings when a simulation is executed several times."
```
```suggestion
    requirement = "The system shall include a graphical interface which maintains its Exodus file viewing settings when a simulation is executed several times."
```

cticenhour commented on python/peacock/tests/utils/tests:
```diff
@@ -1,37 +1,46 @@
     type = PythonUnitTest
     input = test_WidgetUtils.py
     display_required = true
+    requirement = "The system shall include a graphical interface that can create widgets with notably buttons, labels and a progress bar."
```
```suggestion
    requirement = "The system shall include a graphical interface that can create widgets with buttons, labels and a progress bar."
```

cticenhour commented on python/chigger/tests/speed/speed.py:
```diff
@@ -12,8 +12,13 @@
+import mooseutils
 
-filename = os.path.join(os.getenv('MOOSE_DIR'), 'modules', 'phase_field', 'examples', 'grain_growth', 'grain_growth_2D_graintracker_out.e')
+if os.getenv('MOOSE_DIR') == None:
+    mooseutils.mooseError("MOOSE_DIR must be set in this script")
+
+# filename = os.path.join(os.getenv('MOOSE_DIR'), 'modules', 'phase_field', 'examples', 'grain_growth', 'grain_growth_2D_graintracker_out.e')
+filename = os.path.join(os.getenv('MOOSE_DIR'), 'modules', 'phase_field', 'test', 'tests', 'grain_growth', 'gold', 'test_out.e-s005')
```
Why this change, and did you forget to remove the commented out previous line?

GiudGiud replied:
the original input is not included in the repository. it only appears after running the test suite for phase field. I ll remove the previous line

GiudGiud replied:
Removed the previous line and a few commented lines in the input

GiudGiud committed:
Address Casey's review

better English
use "the system shall" consistently
fix some requirements that were missing text

Co-authored-by: Casey Icenhour <cticenhour@gmail.com>

GiudGiud commented on python/chigger/tests/global_data/tests:
```diff
@@ -0,0 +1,36 @@
+    command = plot_global_data.py
+    imagediff = 'plot_global_data.png'
+    display_required = true
+    requirement = "The system shall be able to draw line plots from global data."
```
I ll double check here. I thought I changed because the test actually does field data.

GiudGiud replied:
seems it was global data. not sure what happened.

GiudGiud commented on python/chigger/tests/rotational_extrusion/tests:
```diff
@@ -1,4 +1,6 @@
+  design = 'chigger/index.md'
   [exodus]
     type = ImageDiff
     command = rotational_extrusion.py
```
done, and for the other tests as well

GiudGiud commented:
failure unrelated (torch)

cticenhour commented on python/chigger/tests/line/tests:
```diff
@@ -27,7 +26,7 @@
     imagediff = 'empty.png'
     allowed_linux = 0.935
-    requirement = 'An empty graph shall be a valid chigger widget'
+    requirement = 'An empty graph shall be a valid The system widget'
```
```suggestion
    requirement = 'The system shall allow an empty graph to be a valid widget'
```

Bad find and replace.

cticenhour commented on python/peacock/tests/execute_tab/ExecuteTabPlugin/tests:
```diff
@@ -1,7 +1,10 @@
     type = PythonUnitTest
     input = test_ExecuteTabPlugin.py
     display_required = true
+    requirement = "The system shall include a graphical interface that has a plugin execution tab that lets user specify execution parameters, through the command line, programmatically and on the interface, and set these run option preferences for individual plugins."
```
```suggestion
    requirement = "The system shall include a graphical interface that has a plugin execution tab that lets users specify execution parameters, through the command line, programmatically and on the interface, and set these run option preferences for individual plugins."
```

cticenhour reviewed (changes requested):
A couple more things, but I think after this it is ready to go.

GiudGiud commented:
Failures unrelated, due to current work moving an item from BISON to Moose

GiudGiud commented:
Thanks for the review!
